File: domain/src/main/java/com/ortin/gesturetranslator/domain/repository/SettingsRepository.java
package com.ortin.gesturetranslator.domain.repository;

import com.ortin.gesturetranslator.domain.models.SettingsDomain;

public interface SettingsRepository {
    boolean saveTheme(SettingsDomain settingsDomain);
    SettingsDomain getTheme();

    boolean saveGpu(SettingsDomain settingsDomain);
    SettingsDomain getGpu();

    boolean savePercent(SettingsDomain settingsDomain);
    SettingsDomain getPercent();

    boolean saveSpeedFrameDetection(SettingsDomain settingsDomain);
    SettingsDomain getSpeedFrameDetection();
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/repository/SettingsRepository.kt
package com.ortin.gesturetranslator.domain.repository

import com.ortin.gesturetranslator.domain.models.SettingsDomain

interface SettingsRepository {
    // TODO Сделать функции suspend
    fun saveSettings(settingsDomain: SettingsDomain) : Boolean
    fun getSettings() : SettingsDomain
}
